cluster-autoscaler/proposals/pricing.md

```diff
@@ -0,0 +1,160 @@
+This number can be, in theory, combined with the existing number using a linear combination:
+
+```
+W1 * C/T + W2 * NodeUnFitness
+``` 
+
+NodeUnFitness will be small for nodes that match “overall” to the cluster shape and big for nodes that don’t match there well. 
+ 
+One of the possible (simple) NodeUnFitness implementations can be defined as a ratio between a perfect node for the cluster and the node from the pool. To be more precise:
+
+``` 
+NodeUnFitness = max(preferred_cpu/node_cpu, node_cpu/preferred_cpu)
+``` 
+
+Max is used to ensure that NodeUnfitness is equal to 1 for a perfect node and greater than that for not-so-perfect nodes.
+ 
+While this linear combination sounds cool it is a bit problematic.
+ 
+For small or single pods C/T strongly prefers smaller machines that may not be the best for the overall cluster well-being. For example C/T for a 100m pod with n1-standard-8 is ~80. 
+C/T for n2-standard-2 is 20. Assuming that n1-standard-8 would be a node of choice if 100 node clusters W2*NodeUnFitness would have to be 60 (assuming, for simplicity, W1 = 1).
+N1-standard-2 is only 4 times smaller than n1-standard-8 so W2 = 15.  But then everything collapses with even smaller pod. For a 50milli cpu pod W2 would have to be 30. So it’s bad. C/T is not good for the linear combination. 
+ 
+So we need something better. 
+ 
+We are looking for a pricing function that:
+ 
+* [I1] Doesn’t go 2 times up with a small change in absolute terms (100->50 mill cpu), is more or less constant for small pods. 
+* [I2] Still penalizes node types that have some completely unneeded resources (GPU). 
+ 
+C/T can be stabilized by adding some value X to C and T. Lets call it a big cluster damper. 
+So the formula is like (C+X)/(T+X). 
+ 
+Lets see what happens if X is the cost of running a 0.5 cpu pod and we have 1 pending pod of size 0.1 cpu.
+ 
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2      |            (0.095 + 0.016) / (0.003 + 0.016) | 5.84  |
+| N1-standard-8      |          (0.380 + 0.016) / (0.003 + 0.016)   | 20.84  |
+| N1-standard-2+GPU  |       (0.795 + 0.016) / (0.003 + 0.016)      | 42  |
+ 
+And what if 1.5 cpu.
+ 
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2     |          (0.095 + 0.016) / (0.003*15 + 0.016)| 1.81 |
+| N1-standard-8     |         (0.380 + 0.016) / (0.003*15 + 0.016) |  6.49 |
+| N1-standard-2+GPU |        (0.795 + 0.016) / (0.003*15 + 0.016)  | 13.0  |
+ 
+Slightly better but still hard to combine linearly with NodeUnfitness.
+ 
+Let’s do another trick: NodeUnfitness*(C + X)/(T+X)
+ 
+0.1 cpu request:
+
+| Machine type | Calculation | Rank |
+|--------------|-------|------| 
+| N1-standard-2  |           4 * (0.095 + 0.016) / (0.003 + 0.016) | 23.36 |
+| N1-standard-8  |         1 * (0.380 + 0.016) / (0.003 + 0.016) | 20.84 |
+| N1-standard-2+GPU |  4 * (0.795 + 0.016) / (0.003 + 0.016) | 168.0 |
```

Comment:
You suddenly put values for NodeUnfitness into the example, with no explanation whatsoever where those values come from.